client/settings/payment-method-settings/payment-request-demo-button.js

```diff
@@ -0,0 +1,90 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { React, useEffect } from 'react';
+import { __ } from '@wordpress/i18n';
+import {
+	PaymentRequestButtonElement,
+	useStripe,
+} from '@stripe/react-stripe-js';
+
+/**
+ * Internal dependencies
+ */
+import {
+	usePaymentRequestButtonType,
+	usePaymentRequestButtonSize,
+	usePaymentRequestButtonTheme,
+} from '../../data';
+
+const PaymentRequestDemoButton = ( props ) => {
+	const stripe = useStripe();
+	const [ buttonType ] = usePaymentRequestButtonType();
+	const [ size ] = usePaymentRequestButtonSize();
+	const [ theme ] = usePaymentRequestButtonTheme();
+	const { paymentRequest, setPaymentRequest, setIsLoading } = props;
+
+	// Since this is preview, we don't want the user to open up the browser's payment popup.
+	const disablePaymentAction = ( e ) => {
+		e.preventDefault();
+	};
+
+	// Helper function to convert UI options to pixels in height.
+	const sizeToPx = () => {
+		const sizeToPxMappings = {
+			default: 40,
+			medium: 48,
+			large: 56,
+		};
+		return sizeToPxMappings[ size ] + 'px';
+	};
+
+	useEffect( () => {
+		if ( ! stripe ) {
+			return;
+		}
+
+		// Create a preview for payment button. The label and its total are placeholders.
+		const stripePaymentRequest = stripe.paymentRequest( {
+			country: 'US',
+			currency: 'usd',
+			total: {
+				label: __( 'Total', 'woocommerce-payments' ),
+				amount: 99,
+			},
+			requestPayerName: true,
+			requestPayerEmail: true,
+		} );
+
+		// Check the availability of the Payment Request API.
+		stripePaymentRequest.canMakePayment().then( ( result ) => {
+			if ( result ) {
+				setPaymentRequest( stripePaymentRequest );
+			}
+			setIsLoading( false );
+		} );
+	}, [ stripe, setPaymentRequest, setIsLoading ] );
+
+	if ( paymentRequest ) {
+		return (
+			<PaymentRequestButtonElement
+				onClick={ disablePaymentAction }
```

Comment:
We chatted about this yesterday - setting a `key` prop to force a re-mount seems to resolve the Safari issue, but also causes a slight flash of content.
I didn't include this in my suggestion's branch for that reason.
I asked Stripe for clarification: p1627659822005100-slack-C9976E5MJ
```suggestion
			<PaymentRequestButtonElement
			    key={ `${ buttonType }-${ theme }-${ size }` }
				onClick={ disablePaymentAction }
```

Without `key`:
![2021-07-30 10 37 36](https://user-images.githubusercontent.com/273592/127678303-90f88d23-1f03-4590-9765-c38d694ab48e.gif)


With `key`:
![2021-07-30 10 45 57](https://user-images.githubusercontent.com/273592/127678280-ed1a1acb-d8c4-4a88-a4b7-fe8343d38218.gif)